tests/test_invenio_search.py

```diff
@@ -448,3 +447,40 @@ def test_create_when_indexes_already_exists_with_ignore_existing_true(app):
         search.client.indices.exists("records-bibliographic-bibliographic-v1.0.0")
         is True
     )
+
+
+def test_update_mappings(app):
+    mapping_path = "tests/mock_module/mappings/os-v2/records/default-v1.0.0.json"
+    current_search_client.indices.delete("*")
+
+    with open(mapping_path, "r") as body:
+        initial_mapping = json.load(body)
+
+    current_search_client.indices.create(
+        index="records-default-v1.0.0", body=initial_mapping
+    )
+    current_search.register_mappings("records", "mock_module.mappings")
+    assert current_search_client.indices.exists("records-default-v1.0.0") is True
+
+    # change mapping
+    with open(mapping_path, "w") as mapping_file:
+        # test addition
+        new_mapping = {"mappings": {"properties": {"title": {"type": "keyword"}}}}
+        json_mapping = json.dumps(new_mapping)
+        mapping_file.write(json_mapping)
+
+    current_search.update_mapping("records-default-v1.0.0")
+
+    with open(mapping_path, "w") as mapping_file:
+        # test change
+        new_mapping = {"mappings": {"properties": {"abstract": {"type": "text"}}}}
+        json_mapping = json.dumps(new_mapping)
+        mapping_file.write(json_mapping)
+
+    with pytest.raises(NotAllowedMappingUpdate):
+        current_search.update_mapping("records-default-v1.0.0")
+
+    with open(mapping_path, "w") as mapping_file:
```

Comment:
should this be a fixture: setup, yield, reset ?

Reply:
not sure? we are not using it anywhere else

Reply:
I agree, but you in the fixture you can make a try/except and reset in case of failure. With a test like this, if there is a failure (e.g on an assert) in the middle. Changes won't be undone.